### DISCUSSION TITLE
Override table automatically added to SELECT FROM clause

### DISCUSSION THREAD

**Discussion by mrlubos** [sqlalchemy/sqlalchemy#8486] [Usage Questions] [answered]
Hey! My question is essentially the same one as this one on [StackOverflow](https://stackoverflow.com/questions/24590521/sqlalchemy-override-automatic-from-clause-creation-from-select) but no one answered it there.

In my case, creating a complete statement doesn't help. Here's the offending generated SQL query.

```sql
SELECT a.id,
    (
        SELECT count(*) AS count_1
        FROM (
                SELECT b.col AS col
                FROM b,
                    a -- this shouldn't be here!
                WHERE b.a_id = a.id
                GROUP BY b.col
            ) AS anon_1
    ) AS cols
FROM a
    LEFT OUTER JOIN b ON b.a_id = a.id
GROUP BY a.id
```

And here's the SQLAlchemy statement.

```py
select(
    a.id,
    select(func.count('*')).select_from((
        select(b.col)
        .select_from(b) # why do you insist on selecting from 'a'?
        .where(b.a_id == a.id)
        .group_by(b.col)
    )).label('cols'),
)
.join(b, b.a_id == a.id, isouter=True)
.group_by(a.id)
```

So, the question is, how to get rid of that `a` table from the FROM clause?

**Comment by mrlubos** [comment]
I've also opened an issue as I'm not sure which category this falls into https://github.com/sqlalchemy/sqlalchemy/issues/8487

**Reply by CaselIT** [reply]
There is no need to open both. Maintainers  can move it if needed. I've closed the issue

**Comment by CaselIT** [comment, marked as answer]
Hi,

You are using a in the where, so sqlalchemy adds it to the from clause if the autocorrelation logic does not trigger.
To force correlation on a table you can use `correlate()` or `correlate_except()`

**Reply by mrlubos** [reply]
Thank you!
